### ISSUE TITLE
Window cannot be restored after it has been minimized

### ISSUE THREAD

**Issue by adamwinn** [Sequel-Ace/Sequel-Ace#1416]
- Sequel Ace Version (including build number): Version 3.5.1 Build 20031
- Sequel Ace Source (App Store/GitHub/Homebrew): Github
- macOS Version: 12.2.1 (21D62)
- Processor Type (Intel/Apple): Apple
- MySQL Version: 5.7.12
- macOS Localization: en

**Is Issue Present in [Latest Beta](https://github.com/Sequel-Ace/Sequel-Ace/releases)?**
Yes

**Description**
After connecting to a remote server and viewing the contents of a table, if the window is then minimized, clicking on the icon in the dock will not restore the window

**Steps To Reproduce**
1.   Connect to a remote DB
2.  View the contents of a table
3.  Minimize the window
4.  Click the Sequel Ace icon in the dock to try to restore the window

**Expected Behaviour**
Window is restored after clicking on the icon in the dock

**Related Issues**
#1193 #1194 #1173

**Additional Context**
I cannot reproduce this bug when connecting to localhost
Issue only exists when connecting to a remote server

- Opening Sequel Ace and then minimizing the window - works fine
- Opening Sequel Ace and connecting to a remote DB and then minimizing the window - works fine
- Opening Sequel Ace and connecting to a remote DB and selecting a table and viewing the contents - bug exists

I have done a complete uninstall and remove of all files and preferences with AppCleaner and installed Sequel Ace fresh and the bug exists.

I can reproduce this issue at will at any time.  If you need more info, just let me know.

Example of issue
![My Movie](https://user-images.githubusercontent.com/3028964/157778288-d72eddf8-ab3f-47f4-9b4e-89540b97b151.gif)

**Comment by Jason-Morcos** [comment]
@Kaspik Window-management-related perhaps?

@adamwinn I believe, as a workaround, you can right click and choose the window from the list of windows to restore it, correct?

**Comment by adamwinn** [comment]
Yes, that's what I also showed in the attached video

**Comment by tkafka** [comment]
I would love to have this fixed, Sequel Ace is the only app on my mac where clicking the dock icon doesn't restore the minimized window.

**Comment by Jason-Morcos** [comment]
> I would love to have this fixed, Sequel Ace is the only app on my mac where clicking the dock icon doesn't restore the minimized window.

PRs are welcome and always reviewed! :)

**Comment by Kaspik** [comment]
Is this still an issue with 4.0.0?

**Comment by tkafka** [comment]
@Kaspik Yes, same problem - and it's really weird that the window restore works fine, until I open a table structure or content - and then it stops working.
